test/navigation.spec.js

```diff
@@ -396,29 +395,40 @@ module.exports.describe = function({testRunner, expect, playwright, MAC, WIN, FF
         expect(response.status()).toBe(200);
       });
 
+      /**
+       * @param {import('../src/frames').Frame} frame 
+       * @param {TestServer} server 
+       * @param {'networkidle0'|'networkidle2'} signal 
+       * @param {() => Promise<void>} action 
+       * @param {boolean} isSetContent 
+       */
       async function networkIdleTest(frame, server, signal, action, isSetContent) {
-        let lastResponseFinished;
         const finishResponse = response => {
-          lastResponseFinished = performance.now();
           response.statusCode = 404;
           response.end(`File not found`);
         };
-
+        const waitForRequest = suffix => {
+          return Promise.all([
+            server.waitForRequest(suffix),
+            frame._page.waitForRequest(server.PREFIX + suffix),
```

Comment:
I am curious why do we need to waitForRequest for the test to pass.

Reply:
I don't remember, but it is definitely required. I'll look at it again tomorrow.

Reply:
The protocol events for firefox come in a strange order if we dont await the request events on the frontend:
1. requestWIllBeSent A
2. responseRecieved A
3. requestWillBeSent B
3. requestWillBeSent C

Because A appears to be resolved before B and C are sent, networkidle2 is never trigged, and the timer resolves 500ms after the initial index.html request, not the responseRecieved A as was intended.